lib/splitclient-rb/engine/matchers/matcher.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module SplitIoClient
+  #
+  # class to implement the all keys matcher
+  #
+  class Matcher
+    #
+    # evaluates if the given object equals the matcher
+    #
+    # @param obj [object] object to be evaluated
+    #
+    # @return [boolean] true if obj equals the matcher
+    def equals?(obj)
+      if obj.nil?
```

Comment:
this will be handled by `!obj.instance_of?(self.class)` since `nil` is an instance of `NilClass`